gaps/gap-3_mid_agreement_payments/gap-3_mid_agreement_payments.md

```diff
@@ -0,0 +1,104 @@
+---
+gap: 3
+title: Mid Agreement payments
+description: Specification of Agreement standard for negotiating and describing continues payments during task execution.  
+author: nieznany.sprawiciel (@nieznanysprawiciel), stranger80 (@stranger80), shadeofblue (@shadeofblue)
+status: Draft
+type: Feature
+---
+
+## Abstract
+Introduces Agreement properties standard to allow negotiating incremental payments during task execution.
+
+## Motivation
+Current payments scheme allows only to pay for tasks, after all computations are finished and Invoice is sent to Requestor.
+In case of long-running tasks or services, Provider would expect, that he will be paid during
+execution, because he doesn't want to wait for task finish.
+
+## Specification
+
+Setting `due_date` in Debit Note will cause yagna payment module to automatically pay for Debit Note,
+after it was accepted. This specification describes, how Requestor and Provider should negotiate payments parameters.
+
+### New properties
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| `golem.com.scheme.payu.debit-note-interval-sec?` | Number (uint32) [Negotiable] | Minimum time (in seconds) between subsequent Debit Notes. The Provider must not issue the next Debit Note before the interval period. The Requestor ignores Debit Notes issued before interval elapses. |
+| `golem.com.scheme.payu.payment-timeout-sec?` | Number (uint32) [Negotiable] | Maximum time to receive payment for any Debit Note. This time is counted from Debit Note’s issue date. |
+
+- Requestor can ignore Debit Note, if it came before expected interval
+  - If Requestor is spammed with Debit Notes by Provider, he is allowed to break Agreement from Provider's fault
+- In the mechanism suggested above a Debit Note shall have PaymentDueDate set, and the values must be set so that they correspond with the agreed `payment-interval-sec` value.
+  - Intervals are relative to the agreement timestamp
+- It is up to payment driver implementation to decide, when we can treat Debit Note as paid.
+  - Requestor's payment driver must ensure, that transaction will be confirmed in given timestamp. 
+  - Provider can check `DebitNoteSettledEvent` timestamp to know, if Debit Note was paid in time.
+
+### Deprecation
+
+`golem.com.scheme.payu.interval_sec` - Replaced by negotiable `golem.com.scheme.payu.debit-note.interval-sec?`. Provider
+using mid-agreement payments specification can't set `interval_sec` property.
+
+**Note:** We encourage using `golem.com.scheme.payu.debit-note.interval-sec?` property in new implementations,
+even when parties don't need mid-agreement payments.
+
+### Impact analysis
+
+Following modifications need to be applied to components in Golem suite:
+
+- `yagna`
```

Comment:
should we also include the modifications required for the "nagative paths" i.e. breaking the agreement due to failing to meet agreed times

Reply:
I added validation of time intervals on yapapi/yajsapi side.
I can't see any other cases that weren't covered